test/nostrum/shard/dispatch_test.exs

```diff
@@ -18,5 +18,20 @@ defmodule Nostrum.Shard.DispatchTest do
 
       assert(^result = expected)
     end
+
+    test "returns `MessageDeleteBulk.t()` given Message Delete event" do
+      result =
+        Dispatch.handle_event(
+          :MESSAGE_DELETE_BULK,
+          Stubs.gateway_message_delete_bulk_payload(),
+          %{}
+        )
+
+      expected =
+        {:MESSAGE_DELETE_BULK,
+         struct(MessageDeleteBulk, Stubs.gateway_message_delete_bulk_payload()), %{}}
```

Comment:
It feels a little weird to me to be just throwing the event payload stub into the struct here. I know it's correct since this event isn't doing a whole lot, but I think it can be done better.

How would you feel about changing this to be like the `MessageDelete` event, but instead of the hardcoded keys we can access the fields individually when creating the expected struct? In the future where keys from payloads can be missing or added to events we won't have to worry about how to handle that.
```suggestion
         struct(MessageDeleteBulk, %{
           channel_id: Stubs.gateway_message_delete_bulk_payload().channel_id,
           guild_id: Stubs.gateway_message_delete_bulk_payload().guild_id,
           ids: Stubs.gateway_message_delete_bulk_payload().ids
         }),
         %{}
       }
```
I think this makes it clearer where the keys are coming from. I'd like to hear your thoughts.

Reply:
If a payload needs to be updated due to missing fields or deprecated fields, then I feel it'd be appropriate to update the payload stub.

I don't think readers need to know where the keys are coming from. If my intuition is correct, the test checks that it converts the given payload to its struct counterpart. No part of that intuition requires knowing what keys are included in the payload.

Reply:
On a slightly related note...

The payload stub doesn't cover every situation appropriately. For example, the `guild_id` in the bulk delete's stub can be missing from the payload. The current payload includes a `guild_id`.

It might be appropriate to introduce generators that can correctly produce that kind of data. Something like `StreamData` offers this. That'd allow us to incorporate tests that cover a larger variety of data samples.

Reply:
I think the tests here should test that a given payload results in a proper event sent to the user. There's a lot of nuance here that I'm trying to plan ahead for.

Like for example, when a guild is created we do a bit of indexing on certain fields. In that case our expected result would need to be crafted a bit carefully, unless we have the fields indexed in the initial payload, but that kind of defeats the purpose IMO.

As for the `StreamData` inclusion and property testing, I think we should move this to it's own PR as it might be out of scope here.

Reply:
Marking this as resolved. Cleared confusion on discord.